graspologic/embed/case.py

```diff
@@ -0,0 +1,209 @@
+        < min(Y.shape). If None, then optimal dimensions will be chosen by
+        ``select_dimension`` using ``n_elbows`` argument.
+
+    n_elbows : int, optional, default: 2
+        If `n_components=None`, then compute the optimal embedding dimension using
+        `select_dimension`. Otherwise, ignored.
+
+    check_lcc : bool , optional, defult = True
+        Whether to check if input graph is connected. May result in non-optimal
+        results if the graph is unconnected. Not checking for connectedness may
+        result in faster computation.
+
+
+    References
+    ---------
+    .. [1] Binkiewicz, N., Vogelstein, J. T., & Rohe, K. (2017). Covariate-assisted
+    spectral clustering. Biometrika, 104(2), 361-377.
+    """
+
+    def __init__(
+        self,
+        alpha: Optional[float] = None,
+        assortative: bool = True,
+        n_components: Optional[int] = None,
+        n_elbows: int = 2,
+        check_lcc: bool = False,
+    ):
+        super().__init__(
+            n_components=n_components,
+            n_elbows=n_elbows,
+            check_lcc=check_lcc,
+            concat=False,
+            algorithm="eigsh",
+        )
+
+        if not isinstance(assortative, bool):
+            msg = "embedding_alg must be a boolean value."
+            raise ValueError(msg)
+        self.assortative = assortative
+
+        if not ((alpha is None) or isinstance(alpha, (float, int))):
+            msg = "alpha must be in {None, float, int}."
+            raise TypeError(msg)
+
+        self.alpha = alpha
+        self.latent_right_ = None
+        self.is_fitted_ = False
+
+    def fit(
+        self, network: Tuple[np.ndarray, np.ndarray], y: None = None
+    ) -> "CovariateAssistedEmbed":
+        """
+        Fit a CASE model to an input graph, along with its covariates. Depending on the
+        embedding algorithm, we embed
+
+        .. math:: L_ = LL + \alpha YY^T
+        .. math:: L_ = L + \alpha YY^T
+
+        where :math:`\alpha` is a tuning parameter which makes the leading eigenvalues
+        of the two summands the same. Here, :math:`L` is the regularized
+        graph Laplacian, and :math:`Y` is a matrix of covariates for each node.
+
+        Covariates are row-normalized to unit l2-norm.
+
+        Parameters
+        ----------
+        network : tuple or list of np.ndarrays
+            Contains the tuple (A, Y), where A is an adjacency matrix and Y is the
+            matrix of covariates.
+
+            A : array-like or networkx.Graph
+                Input graph to embed. See graspologic.utils.import_graph
+
+            Y : array-like, shape (n_vertices, n_covariates)
+                Covariate matrix. Each node of the graph is associated with a set of
+                `d` covariates. Row `i` of the covariates matrix corresponds to node
+                `i`, and the number of columns are the number of covariates.
+
+        y: Ignored
```

Comment:
It was actually originally A, Y a few weeks ago. There are three reasons I changed it:

1. to match the syntax in MASE and OMNI, which both eat iterables of graphs
2. because I'd have to add  `*args, **kwargs` to the base class's `fit_transform` method to get it to eat two arguments, which felt unnecessarily complicated
3. because in the book, jovo has been pushing that a "network" is a single object which contains both the topological information (adjacency matrix) and the covariates, and whatever other extra information is associated with the graph. Passing a single object as input felt more harmonious with this idea.

If you're really attached to not using a tuple as input, I can change it back

Reply:
name change from `A` to `graph` and `Y` to `covariates`: 164dbb4

Reply:
@j1c @asaadeldin11 second opinion on this? the tuple feels weird to me still, for omni/mase i think it makes more sense as there are an arbitrary number of inputs and they are essentially unordered, here there are always two and each has a specific meaning.

Reply:
Wouldn't a tuple make **more** sense for ordered arguments, since tuples are immutable?

Reply:
by that logic then all functions should accept one tuple and one dictionary. To me this syntax would be more confusing than just the two named arguments but want to see what @j1c or @asaadeldin11 think

Reply:
I think it makes sense to have separate args, and it doesn't make sense for input graph and covariates to be ordered.

Reply:
ok, changed back to separate args: 18092b0